File: frontend/src/types/wiki.ts
export interface PageNode {
    id: number;
    title: string;
    summary: string;
}

export interface WikiPath {
    start_page: PageNode;
    end_page: PageNode;
    path: PageNode[];
}

export interface WikiNode {
    id: number;
    title: string;
    summary: string;
}

export interface WovenStory {
    nodes: WikiNode[];
    story?: {
        content: string;
    };
}

File: api/models/wiki.py
from pydantic import BaseModel
from typing import List, Optional


class PageNode(BaseModel):
    id: int
    title: str
    summary: str


class WikiPath(BaseModel):
    start_page: PageNode
    end_page: PageNode
    path: List[PageNode]


class WikiNode(BaseModel):
    id: int
    title: str
    summary: str


class WovenStory(BaseModel):
    nodes: List[WikiNode]
    story: Optional[dict] = None
